File: 백준/Silver/1966. 프린터 큐/프린터 큐.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int TC = Integer.parseInt(br.readLine());
        while (TC-- > 0) {
            String[] input = br.readLine().split(" ");
            int N = Integer.parseInt(input[0]);
            int M = Integer.parseInt(input[1]);

            LinkedList<int[]> q = new LinkedList<>();
            String[] nums = br.readLine().split(" ");
            for (int i=0; i<nums.length; i++) {
                q.add(new int[]{i, Integer.parseInt(nums[i])});  // 인덱스, 우선순위
            }

            int answer = 0;
            while (!q.isEmpty()) {
                int[] cur = q.pop();
                boolean isMax = true;
                for (int i=0; i<q.size(); i++) {
                    if (cur[1] < q.get(i)[1]) {
                        isMax = false;
                        q.add(cur);
                        break;
                    }
                }
                if (isMax) {
                    answer++;
                    if (cur[0] == M) {
                        System.out.println(answer);
                        break;
                    }
                }
            }
        }
    }
}

File: 백준/Silver/1966. 프린터 큐/프린터 큐.py
for _ in range(int(input())):
    n, idx = map(int, input().split())    # n: 프린트 수 m: 대상 인덱스
    printer = list(map(int, input().split()))   # 우선순위
    cnt = 0

    while len(printer) > 0:
        if idx > 0:
            if max(printer) == printer[0]:
                printer.pop(0)
                cnt += 1
                idx -= 1
            else:
                printer.append(printer.pop(0))
                idx -= 1
        else:
            if max(printer) == printer[0]:
                cnt += 1
                print(cnt)
                break
            else:
                printer.append(printer.pop(0))
                idx = len(printer) - 1